base/deprecated.jl

```diff
@@ -34,6 +35,16 @@ old (generic function with 1 method)
 julia> @deprecate old(x) new(x) false
 old (generic function with 1 method)
 ```
+
+To deprecate a specific signature, annotate the arguments of `old`. For example,
```

Comment:
When you write "deprecate a specific signature" it sounds like you mean deprecating one signature and not another (so the example feels backward to me). I.e. this example
```
f(x::Int) = #...
f(x::Float64) = # ...
```
and you want to deprecate `f(::Int)` but not `f(::Float64)` and do
```
@deprecate f(x::Int) f(float(x))
```

Reply:
My intention was to say that there previously existed some API `old(::Integer)` which has now be supplanted by `new(::Integer)`. Let me see if I can rephrase it. (In any case, I'll reduce it to a single-argument method as you suggested)

Reply:
Yea I undrerstood what you meant. Perhaps something like

> Note that by default methods accepting arguments of typ `Any` is defined, if you want to restrict the deprecated method use ...

Reply:
I tried a variation on your suggestion - see latest commit.